check_pedigree.py

```diff
@@ -0,0 +1,1217 @@
+    :param vcf_samples: dict
+    :return: trios given in the pedigree, duos given in the pedigree, and dictionary of each sample's family ID, dictionary of project IDs for each sample
+    :rtype: Tuple[hl.Pedigree, list, dict, dict]
+    """
+
+    ped_trios = []
+    ped_duos = []
+
+    given_families = {}
+    seqr_projects = defaultdict(str)
+    
+    
+    out_new_ped = hl.hadoop_open("{output_dir}/{output_name}_functioning_pedigree.ped".format(**locals()),'w')
+    
+    
+    with hl.hadoop_open(input_pedigree, 'r') as infile:
+        next(infile)
+        for line in infile:
+            line = line.rstrip('\n')
+            items = line.split('\t')
+            Project, Family_ID, Individual_ID, Paternal_ID, Maternal_ID, Sex, Affected = items[0:7]
+        
+            given_families[Individual_ID] = Family_ID
+        
+            if Individual_ID not in vcf_samples:
+                Individual_ID = "."
+            if Paternal_ID not in vcf_samples:
+                Paternal_ID = "."
+            if Maternal_ID not in vcf_samples:
+                Maternal_ID = "."
+            
+            
+            # remove line from pedigree if the proband is missing
+            if Individual_ID != ".":
+                seqr_projects[Individual_ID] = Project
+                out_new_ped.write("{Family_ID}\t{Individual_ID}\t{Paternal_ID}\t{Maternal_ID}\t{Sex}\t{Affected}\n".format(**locals()))
+         
+            # for comparision purposes to ibd output, set sex and family to none (sex is already checked in a separate step)
+            Sex = None
+            Family_ID = None
+         
+            if Individual_ID in vcf_samples:
+                if Paternal_ID in vcf_samples and Maternal_ID in vcf_samples:
+                    ped_trios.append(hl.Trio(Individual_ID, Family_ID, Paternal_ID, Maternal_ID,Sex))
+                elif Paternal_ID in vcf_samples and Maternal_ID not in vcf_samples:
+                    ped_duos.append(sorted([Individual_ID,Paternal_ID]))
+                elif Paternal_ID not in vcf_samples and Maternal_ID in vcf_samples:
+                    ped_duos.append(sorted([Individual_ID,Maternal_ID]))
+                
+
+    given_trios = hl.Pedigree(ped_trios)
+    out_new_ped.close()
+    
+    return(given_trios, ped_duos, given_families, seqr_projects)
+
+
+
+def annotate_parent_decisions(
+    trios: hl.Pedigree,
+    duos: list,
+    ht: hl.Table
+) -> hl.Table:
+
+    """
+    Annotates who the parents are for trios and duos
+    :param input_pedigree: str
+    :param vcf_samples: dict
+    :return: ht with annotated decision columns for who the parent(s) of the sample were
+    :rtype: hl.Table
+    """
+    
+    # filter out confirmed siblings
+    parent_decisions = {}
+
+    for trio in trios:
+        parent_decisions[trio.s] = [trio.pat_id, trio.mat_id]  # two parents
+    
+    for duo in duos:
+        s1 = duo[0]
+        s2 = duo[1]
+        parent_decisions[s1] = [str(s2)]  # duos
```

Comment:
here, it seems like you annotate things both ways (offspring -> parent and parent -> offspring) whereas in the trio case it's only offspring -> parent.